Realm/RLMConstants.h

```diff
@@ -18,6 +18,14 @@
 
 #import <Foundation/Foundation.h>
 
+// For compatibility with Xcode 7, before extensible string enums were introduced,
+#ifdef NS_EXTENSIBLE_STRING_ENUM
```

Comment:
I'm not confident this is sufficient. If the `NS_EXTENSIBLE_STRING_ENUM` works in Swift 2.x when compiled with Xcode 8, then this will break the API on upgrade from Xcode 7 to Xcode 8 even if the developer uses a legacy Swift version. I'm not sure if this is problematic though, as it's a simple rename and would only break on upgrade.

I would definitely appreciate hearing others' thoughts on this though because I'm not confident I considered all possibilities.